### PR TITLE
✨ defines openapi specs for new studies resources

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Drafts the openapi specs (OAS) for a new ``studies`` collections and missing operations in current resources. NOTE that this PR only addresses the OAS, i.e. all handlers will raise ``NotImplementeError`` if queried and are only active in with `API_SERVER_DEV_FEATURES_ENABLED=1`

The idea of this PR is to draft the interface and create tests for the issues requested so far as a starting point for its implementation.

### Highlights
- ✨  OAS for ``studies`` collections
- ✨  OAS missing operations in current resources (see related issues)
-  🔨 ``make openapi-dev.json`` produces OAS with `API_SERVER_DEV_FEATURES_ENABLED=1`

## Related issue/s

- related to https://github.com/ITISFoundation/osparc-simcore/issues/4177: defines study OAS
- related to https://github.com/ITISFoundation/osparc-simcore/issues/4111: defines delete job OAS

## How to test

- prepare tests for implementation
- Can render new OAS. This PR includes (temporary) ``openapi-dev.json`` so the review can explore the new definitions using https://petstore.swagger.io/?url=https://raw.githubusercontent.com/ITISFoundation/osparc-simcore/b95953661d79408f2ef88b100ed2b2a9ed3b7b97/services/api-server/openapi-dev.json

## Follow up

- @bisgaard-itis can you try to create the client?
- @sanderegg I would like to check with you how to connect some of the implementations

### CODE REVIEW

**Review by bisgaard-itis — APPROVED** [review]
Looks good to me. Just a single suggestion and a question.

File: services/api-server/Makefile [hunk 1]
```diff
@@ -84,20 +84,30 @@ run-fake-devel: # starts a fake server in a dev-container
...
 	@cd $(CURDIR); \
 	$(SCRIPTS_DIR)/openapi-generator-cli.bash validate --input-spec /local/$@
+endef
+
+
+.PHONY: openapi-specs openapi.json openapi-dev.json
+
+openapi-specs: openapi.json
+
+openapi.json: .env
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/api-server/Makefile:106]: .env is generated from .env-devel and provides a testing environment 
Yes, you are right that LOG_FORMAT_LOCAL_DEV_ENABLED needs to be moved there. 
in the case of  API_SERVER_DEV_FEATURES_ENABLED I leave it outside so I can override it for `openapi-dev.json`

[the remark this answers is not in the document]
Comment (bisgaard-itis) [reply services/api-server/Makefile:105]: Thanks for the clarification.

File: services/api-server/.env-devel [hunk 2]
```diff
@@ -13,6 +13,7 @@
 #
 
 API_SERVER_DEV_FEATURES_ENABLED=1
+LOG_FORMAT_LOCAL_DEV_ENABLED=1
```
Comment (GitHK) [inline services/api-server/.env-devel:16]: I would advise agains this one. If you require logs in the `full format` just add the env var in the project's root folder `.env` file.
This file is also captured by the osparc autoteplyer and interpolated when it composes the env vars. That's why I would not add it here. Let's avoid unwanted behaviours.

Comment (pcrespov) [reply services/api-server/.env-devel:16]: - this .env-devel is only use for testing/development in ``api-server` only. NOTE that this is NOT the .env-devel of the base!
- `LOG_FORMAT_LOCAL_DEV_ENABLED` was introduced about a month ago after refactoring logs system (@matusdrobuliak66)

**Review by matusdrobuliak66 — APPROVED** [review]

File: services/api-server/.env-devel (hunk 2 above)
Comment (pcrespov) [reply services/api-server/.env-devel:16]: yes

**Review by sanderegg — APPROVED** [review]

**Review by GitHK — APPROVED** [review]

**Pull request merged**